utils/CMakeLists.txt

```diff
@@ -2,8 +2,12 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+set(nrfx_DIR "${iree-bare-metal-arm_SOURCE_DIR}/third_party/nrfx/")
+
 if(BUILD_WITH_CMSIS)
-  if(ARM_TARGET MATCHES "^STM32F4")
+  if(ARM_TARGET STREQUAL "NRF52840")
+    set(UTILS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/nrf52840_cmsis.c ${nrfx_DIR}/drivers/src/nrfx_uart.c)
```

Comment:
Let's create a separate target for `nrfx_uart.c` in `third_party/nrfx`. That can be added as a conditional dep to the `utils` target. I can take over here and add this to your PR :)

Reply:
@lucas-camp Feel free to cherry-pick https://github.com/iml130/iree-bare-metal-arm/commit/c41a53a2da303cd99309203cb4bd79973417f308 and apply it to your PR.